Added Universal Code Search blog post for GitLab (#662)
sourcegraph/about: blogposts/universal-code-search-gitlab.md

---
title: 'Universal Code Search for GitLab'
author: Ryan Blunden
authorUrl: https://twitter.com/ryan_blunden
publishDate: 2020-03-12T05:59-06:00
tags: [
  blog
]
slug: universal-code-search-gitlab
heroImage: /external-logos/gitlab-mark.svg
published: true
---

Sourcegraph [Universal Code Search](/universal-code-search) enables developers to explore and better understand all code, faster, with contextual code intelligence.

> This screencast (with accompanying audio) shows how to [connect and configure](https://docs.sourcegraph.com/admin/external_service/gitlab) Sourcegraph Universal Code Search for public and private repositories on GitLab.com and GitLab CC/EE instances

Reviewer: ```suggestion
This screencast (with accompanying audio) shows how to [connect and configure](https://docs.sourcegraph.com/admin/external_service/gitlab) Sourcegraph Universal Code Search for public and private repositories on GitLab.com and GitLab CE/EE instances
```

Reviewer: Perhaps this is more clear in the screencast, but there's a big difference between enabling for my account on GitLab.com and all the other configuration/admin steps when running your own GitLab instance. Can you make sure to distinguish between the two here?

Author: Good call re GitLab.com vs. GitLab self-hosted. I'll change the wording so it's specific to CE/EE.

> In addition to connection configuration, it discusses important factors for determining which repositories Sourcegraph will index, such as excluding archived repositories, as well as repositories based on naming patterns such `deploy-`.

Reviewer: ```suggestion
In addition to configuring the Sourcegraph-GitLab connection, the screencast discusses important factors for setting up your Sourcegraph instance, such as determining which repositories Sourcegraph will index, excluding archived repositories, or including repositories based on naming patterns (e.g.,`deploy-`).
```

Author: I incorporated most of these changes, thanks!

<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/397320487?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://www.youtube.com/watch?v=r8kvCUeeMKQ" target="_blank">Watch on YouTube</a></p>
</p>

See the [Sourcegraph GitLab documentation](https://docs.sourcegraph.com/admin/external_service/gitlab) for full configuration options, including [enabling repository permissions](https://docs.sourcegraph.com/admin/external_service/gitlab#repository-permissions).

> The script used in the screencast to search for repositories to ignore is available as a [GitLab snippet](https://gitlab.com/snippets/1952534).

Reviewer: This reads awkwardly but I don't have an improvement. There are some typos in the comment of your script, and I think there's a stray 'f' on line 14.

Author: Thanks, fixed the typos in the script and I'll re-work this sentence.

> The script used in the screencast to search for repositories to ignore is available as a [GitLab snippet](https://gitlab.com/snippets/1952534).

Reviewer 2: ```suggestion
You can use this [Python script](https://gitlab.com/snippets/1952534) used in the screencast to search for repositories you want to ignore.
```

Author: I took your idea and incorporated it into a new version of this sentence. Thanks!
